tests/contrib/langchain/test_langchain.py

```diff
@@ -1256,3 +1258,93 @@ def test_vectorstore_logs_error(langchain, ddtrace_config_langchain, mock_logs,
             "documents": [],
         }
     )
+
+
+@pytest.mark.parametrize("ddtrace_global_config", [dict(_llmobs_enabled=True, _llmobs_sample_rate=1.0)])
+class TestLLMObsLangchain:
+    @staticmethod
+    def _expected_llmobs_calls(span, input_role=None, output_role=None):
+        input_meta = {"content": mock.ANY}
+        if input_role is not None:
+            input_meta["role"] = input_role
+
+        output_meta = {"content": mock.ANY}
+        if output_role is not None:
+            output_meta["role"] = output_role
+
+        expected_llmobs_writer_calls = [mock.call.start()]
+        expected_llmobs_writer_calls += [
+            mock.call.enqueue(
+                _expected_llmobs_llm_span_event(
+                    span,
+                    model_name=span.get_tag("langchain.request.model"),
+                    model_provider=span.get_tag("langchain.request.provider"),
+                    input_messages=[input_meta],
+                    output_messages=[output_meta],
+                    parameters={
+                        "temperature": float(span.get_tag("langchain.request.openai.parameters.temperature")),
+                        "max_tokens": int(span.get_tag("langchain.request.openai.parameters.max_tokens")),
+                    },
+                    token_metrics={
+                        "prompt_tokens": span.get_metric("langchain.tokens.prompt_tokens"),
+                        "completion_tokens": span.get_metric("langchain.tokens.completion_tokens"),
+                        "total_tokens": span.get_metric("langchain.tokens.total_tokens"),
+                    },
+                    tags={
+                        "ml_app": "unnamed-ml-app",
+                        "service": "",
+                    },
+                )
+            )
+        ]
+        return expected_llmobs_writer_calls
+
+    @classmethod
+    def _test_llmobs_invoke(
+        cls,
+        generate_trace,
+        request_vcr,
+        mock_llmobs_writer,
+        mock_tracer,
+        cassette_name,
+        input_role=None,
+        output_role=None,
+    ):
+        # disable the service before re-enabling it, as it was enabled in another test
+        LLMObs.disable()
+        LLMObs.enable(tracer=mock_tracer)
+
+        if sys.version_info < (3, 10, 0):
+            cassette_name = cassette_name.replace(".yaml", "_39.yaml")
+
+        with request_vcr.use_cassette(cassette_name):
+            generate_trace("Can you explain what an LLM chain is?")
+        span = mock_tracer.pop_traces()[0][0]
+
+        expected_llmobs_writer_calls = cls._expected_llmobs_calls(span, input_role, output_role)
+        assert mock_llmobs_writer.enqueue.call_count == 1
+        mock_llmobs_writer.assert_has_calls(expected_llmobs_writer_calls)
+
+    def test_llmobs_llm(self, langchain, mock_llmobs_writer, mock_tracer, request_vcr):
```

Comment:
Could we add some additional test coverage for more model providers? I.e. cohere, ai21) similar to how we do it in `test_bedrock.py --> TestLLMObsBedrock`? Thanks!